interface/JetsProducer.h

```diff
@@ -41,8 +65,30 @@ class JetsProducer: public CandidatesProducer<pat::Jet>, public BTaggingScaleFac
 
         // Tokens
         edm::EDGetTokenT<std::vector<pat::Jet>> m_jets_token;
+        edm::EDGetTokenT<double> m_rho_token;
 
         std::map<std::string, std::vector<float>*> m_btag_discriminators;
+        // regression stuff
+        bool applyRegression = false;
+        std::string regressionFile = "";
+        TMVA::Reader *bjetRegressionReader = new TMVA::Reader();
```

Comment:
You have leak here since it's not delete. I'd suggest to move the `new` part in the constructor and to switch to a `std::unique_ptr<TMVA::Reader>` to not deal with the deletion.